ocw/lib/openqa.py

```diff
@@ -0,0 +1,79 @@
+from urllib.parse import urlparse
+import requests
+from requests.exceptions import RequestException
+
+# pylint: disable=import-error
+from openqa_client.client import OpenQA_Client
+from openqa_client.const import JOB_STATE_CANCELLED
+from openqa_client.exceptions import OpenQAClientError
+
+
+TAG_IGNORE = "pcw-ignore"
+
+
+def get_url(server):
+    if urlparse(server).scheme != "":
+        return server
+    for scheme in ("https", "http"):
+        try:
+            url = f"{scheme}://{server}"
+            got = requests.get(url, timeout=30)
+            got.raise_for_status()
+            return url
+        except RequestException:
+            pass
+    raise OpenQAClientError(f"Could not connect to server {server}")
+
+
+class OpenQA:
+    __servers = {}
+
+    def __new__(cls, **kwargs):
+        if kwargs["server"] not in cls.__servers:
+            server = get_url(kwargs["server"])
+            # Use same object for "https://openqa.suse.de", "openqa.suse.de" & "https://openqa.suse.de/"
+            for key in (server, urlparse(server).netloc, kwargs["server"]):
+                if key in cls.__servers:
+                    return cls.__servers[key]
+            self = super().__new__(cls)
+            for key in (server, urlparse(server).netloc, kwargs["server"]):
+                cls.__servers[key] = self
+            self.server = server
+        return cls.__servers[kwargs["server"]]
+
+    def __init__(self, **kwargs):
+        kwargs.pop("server")
+        self.__client = OpenQA_Client(server=self.server, **kwargs)
+
+    def is_cancelled(self, job: str) -> bool:
+        if not job.isdigit():
+            raise ValueError(f"Invalid job: {job}")
+        try:
+            status = self.__client.openqa_request('GET', f'jobs/{job}')
+            # Return true if job is either cancelled or done
+            return status['job']['state'] == JOB_STATE_CANCELLED
+        except (TypeError, KeyError, OpenQAClientError):
+            pass
+        return False
+
+
+def is_cancelled(tags) -> bool:
+    if TAG_IGNORE in tags:
+        return False
+    job = tags.get("openqa_var_JOB_ID")
+    if job is None:
+        # Parse "11555642 openqa_created_by=openqa.suse.de/t11555642 openqa_ttl=11100"
+        for tag in ('openqa-cli-test-tag', 'openqa-cli-test-label'):
+            try:
+                job = tags.get(tag).split()[0]
+                break
+            except AttributeError:
+                continue
+    if job is None:
+        return False
+    server = tags.get("openqa_var_SERVER", "https://openqa.suse.de")
```

Comment:
Again this whole flow should work ONLY if all pieces are presented . If something is missing we just return False . And fallback to our usual flow 
```suggestion
    server = tags.get("openqa_var_SERVER")
```